dev/AppLifecycle/AppInstance.cpp

```diff
@@ -25,41 +25,37 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
 
     std::tuple<std::wstring, std::wstring> ParseCommandLine(const std::wstring& commandLine)
     {
-        auto argsStart = commandLine.rfind(c_argumentPrefix);
-        if (argsStart == std::wstring::npos)
+        int argc{};
+        wil::unique_hlocal_ptr<PWSTR[]> argv{ CommandLineToArgvW(commandLine.c_str(), &argc) };
+
+        // Search for ----ms-protocol:
+        for (int index = 0; index < argc; index++)
         {
-            return { L"", L"" };
-        }
+            std::wstring_view fullArgument = argv[index];
+            auto protocolQualifier = wil::str_printf<std::wstring>(L"%s%s%s", c_argumentPrefix, c_protocolArgumentString, c_argumentSuffix);
 
-        // Push past the '----' commandline argument prefix.
-        argsStart += 4;
+            auto argStart = fullArgument.find(protocolQualifier);
+            if (argStart == std::wstring::npos)
+            {
+                continue;
+            }
 
-        auto argsEnd = commandLine.find_first_of(L' ', argsStart);
+            // Push past the '----' commandline argument prefix.
+            argStart += 4;
```

Comment:
nit: maybe constrexpr a null-term char array with the four dashes since used elsewhere and then constrexpr a size variable?  Seems like over-kill for a dead-obviously correct number (4).